diff --git a/node_modules/eslint-plugin-jsdoc/dist/eslint/getJSDocComment.js b/node_modules/eslint-plugin-jsdoc/dist/eslint/getJSDocComment.js
index e091927..2eaa4b5 100644
--- a/node_modules/eslint-plugin-jsdoc/dist/eslint/getJSDocComment.js
+++ b/node_modules/eslint-plugin-jsdoc/dist/eslint/getJSDocComment.js
@@ -65,6 +65,7 @@ const getReducedASTNode = function getReducedASTNode(node, sourceCode) {
     case 'TSEnumDeclaration':
     case 'ClassDeclaration':
     case 'FunctionDeclaration':
+    case 'TSDeclareFunction':
       return looksLikeExport(parent) ? parent : node;
 
     case 'ClassExpression':
diff --git a/node_modules/eslint-plugin-jsdoc/dist/iterateJsdoc.js b/node_modules/eslint-plugin-jsdoc/dist/iterateJsdoc.js
index 1b66839..f431945 100644
--- a/node_modules/eslint-plugin-jsdoc/dist/iterateJsdoc.js
+++ b/node_modules/eslint-plugin-jsdoc/dist/iterateJsdoc.js
@@ -682,7 +682,8 @@ function iterateJsdoc(iterator, ruleConfig) {
       return {
         ArrowFunctionExpression: checkJsdoc,
         FunctionDeclaration: checkJsdoc,
-        FunctionExpression: checkJsdoc
+        FunctionExpression: checkJsdoc,
+        TSDeclareFunction: checkJsdoc
       };
     },
 
