diff --git a/jquery.mousewheel.js b/jquery.mousewheel.js
index 3eadb7edfd212d915c566e15a37768ac7292a32a..fc8ec770aec60bf485029091b369b5ec4801b5a7 100644
--- a/jquery.mousewheel.js
+++ b/jquery.mousewheel.js
@@ -81,11 +81,11 @@
 
     $.fn.extend({
         mousewheel: function(fn) {
-            return fn ? this.bind('mousewheel', fn) : this.trigger('mousewheel');
+            return fn ? this.on('mousewheel', fn) : this.trigger('mousewheel');
         },
 
         unmousewheel: function(fn) {
-            return this.unbind('mousewheel', fn);
+            return this.off('mousewheel', fn);
         }
     });
 
